### ISSUE TITLE
How to save the sparsified model?

### ISSUE THREAD

**Issue by chengwuxinlin** [athglentis/feather#1]
Hello, thanks for sharing your code out, I have a question regarding your code

`import torch
from sparse_utils import Pruner

train_loader = ...
epochs = ...
model = ...
# create a Pruner class instance
pruner = Pruner(model, device=..., final_rate=..., nbatches=..., epochs=...)
optimizer = ...
loss_fn = ...
for epoch in range(epochs):  
    for data, target in train_loader:
        # update the pruning threshold based on the iteration number and the scheduler used
        pruner.update_thresh()    
        output = model(data)
        loss = loss_fn(output, target)
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
    # update the pruning threshold after last step of the optimizer
    pruner.update_thresh(end_of_batch=True)

# finalize sparse model
pruner.desparsify()`

To save the sparsified model, do I just need to save the `model `?

**Comment by chengwuxinlin** [comment]
I also noticed the testing set inference time didn't  change during the training even when the pruning rate was 99

**Comment by athglentis** [comment]
Hello, thanks for your interest in our work! I updated `pruner.desparsify()` and now after it is called the sparse model can be saved by saving the `model` (as now included in the code).

Regarding your second comment, indeed the inference time won't change because PyTorch by default doesn't take advantage of the sparsity in the weight tensors for optimizing the computations. For that you will need to use some library (and perhaps appropriate hardware) that supports sparse matrix operations to accelerate inference using the (unstructured) sparse model. Unfortunately this is outside the scope of this work, we just provide the method for sparsifying the model.

**Comment by chengwuxinlin** [comment]
Thank you so much for the reply, it answered my question.
